radix-engine/src/engine/system_api.rs

```diff
@@ -71,8 +71,13 @@ pub trait SystemApi {
     /// Removes an RENode and all of it's children from the Heap
     fn drop_node(&mut self, node_id: RENodeId) -> Result<HeapRENode, RuntimeError>;
 
+    /// Allocates a new node id useable for create_node
+    fn allocate_node_id(&mut self, node_type: RENodeType) -> Result<RENodeId, RuntimeError>;
+
     /// Creates a new RENode
-    fn create_node(&mut self, re_node: RENode) -> Result<RENodeId, RuntimeError>;
+    /// TODO: Remove, replace with lock_substate + get_ref_mut use
+    fn create_node(&mut self, node_id: RENodeId, re_node: RENode)
+        -> Result<RENodeId, RuntimeError>;
```

Comment:
Understand you have a TODO there, but is there reason we return `RENodeId`?

Reply:
More to keep interface there once we implement local handles. Would you rather it just be removed?